src/credentials/CHIPCert.h

```diff
@@ -326,14 +322,10 @@ struct ChipCertificateData
     BitFlags<KeyUsageFlags> mKeyUsageFlags;     /**< Certificate key usage extensions flags. */
     BitFlags<KeyPurposeFlags> mKeyPurposeFlags; /**< Certificate extended key usage extensions flags. */
     uint8_t mPathLenConstraint;                 /**< Basic constraint: path length. */
-    struct
-    {
-        const uint8_t * R; /**< Pointer to the R element of the signature, encoded as ASN.1 DER Integer. */
-        uint8_t RLen;      /**< Length of R. */
-        const uint8_t * S; /**< Pointer to the S element of the signature, encoded as ASN.1 DER Integer. */
-        uint8_t SLen;      /**< Length of R. */
-    } mSignature;          /**< Certificate signature structure. */
-    uint8_t mTBSHash[chip::Crypto::kSHA256_Hash_Length]; /**< Certificate TBS hash. */
+    const uint8_t * mSignature;                 /**< Pointer to the certificate signature. */
+    uint8_t mSignatureLen;                      /**< Certificate signature length. */
```

Comment:
```suggestion
    size_t mSignatureLen;                      /**< Certificate signature length. */
```

There are probably more bytes lost to padding in this struct than are gained by attempts for thrift by using uint8_t for lengths.  Why not just use size_t for the lengths?  It's guaranteed not to break.

Reply:
What are the benefits of using `size_t` vs `uint8_t` in this context? The use of `uint` values matches the implementation style in this file. If we want to change that probably it makes sense to do that everywhere.

Reply:
size_t is by definition wide enough to index to any location in memory.  The benefit of using size_t is that none of authors, reviewers or maintainers need to worry about whether the length variable is wide enough to hold all possible values.

Using uint8_t is an optimization.  Is it a useful optimization?  If this optimization is actually needed, why hasn't the struct also been reordered to minimize padding?

Reply:
Very good point! I am adding TODO here to follow up. Already working on a separate PR to replace many of these components with ByteSpan and FixedByteSpan types.